core/broadleaf-framework/src/main/java/org/broadleafcommerce/core/order/service/ProductOptionValidationServiceImpl.java

```diff
@@ -67,7 +67,9 @@ public Boolean validate(ProductOption productOption, String value) {
             throw new RequiredAttributeNotProvidedException(message, attributeName);
         } else {
             String validationString = productOption.getValidationString();
+            validationString = Boolean.parseBoolean(environment.getProperty("exploitProtection.xssEnabled", "false")) ? ESAPI.encoder().decodeForHTML(validationString) : validationString;
```

Comment:
Let's pull the environment out to a property with an @Value annotation.  I believe environment.getProperty will force a propagated call to resolve the property each time.  So a DB call, property check, etc.  The exploitProtection.xssEnabled should not be changing at runtime so a once and done at class initialization should be good.